ocaml/typing/typecore.ml

```diff
@@ -3588,7 +3588,7 @@ let rec approx_type env sty =
   match sty.ptyp_desc with
   | Ptyp_arrow (p, ({ ptyp_desc = Ptyp_poly _ } as arg_sty), sty) ->
       (* CR layouts v5: value requirement here to be relaxed *)
-      let p = Typetexp.transl_label p in
+      let p = Typetexp.transl_label p (Some arg_sty) in
```

Comment:
I'm kind of confused by the location of the original comment in https://github.com/ocaml-flambda/flambda-backend/pull/1589#discussion_r1273868439 . `arg_sty` is never going to be a `src_pos` extension in the `ptyp_desc = Ptyp_poly` case - The comment is meant to refer to the general case below, right?

In that case, I agree and am fixing it now

Reply:
Safer and more future-proof to pass in the type here, yes. This all looks good now.